jwst/pipeline/calwebb_tso3.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import unicode_literals, absolute_import
+
+import os
+
+from astropy.table import vstack
+
+from ..stpipe import Pipeline
+from .. import datamodels
+
+from ..outlier_detection import outlier_detection_tso_step
+from ..outlier_detection import outlier_detection_scaled_step
+from ..tso_photometry import tso_photometry_step
+from ..extract_1d import extract_1d_step
+from ..white_light import white_light_step
+
+__version__ = "0.7.8"
+
+
+class TSO3Pipeline(Pipeline):
+    """
+    TSO3Pipeline: Applies level 3 processing to TSO-mode data from
+                    any JWST instrument.
+
+    Included steps are:
+        outlier_detection
+        tso_photometry
+        extract_1d
+        white_light
+    """
+
+    spec = """
+        suffix = string(default='i2d')
```

Comment:
I'm not sure we need or want to have a single "standard" suffix defined for the output from this pipeline. Unlike pipelines like calspec2 or image2 where you know you're always going to at least get a _cal product out the backend, with others possible, there isn't one single product that will always be produced by this pipeline, because it's handling both images and spectra. The only one you should always get are the level-2c (crf) products from outlier_detection, but I think those are saved by outlier_detection itself? From there on, you might get an imaging photometry catalog (_phot product) or you might get an _x1d and _whtlt product.